examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
+	}
+})";
+#endif
+
+#define LOG(...) printf(__VA_ARGS__); printf("\n");
+class DemoEventCallback : public nbl::ui::IWindow::IEventCallback
+{
+	bool onWindowShown_impl() override
+	{
+		LOG("Window Shown");
+		return true;
+	}
+	bool onWindowHidden_impl() override
+	{
+		LOG("Window hidden");
+		return true;
+	}
+	bool onWindowMoved_impl(int32_t x, int32_t y) override
+	{
+		LOG("Window window moved to { %d, %d }", x, y);
+		return true;
+	}
+	bool onWindowResized_impl(uint32_t w, uint32_t h) override
+	{
+		LOG("Window resized to { %u, %u }", w, h);
+		return true;
+	}
+	bool onWindowMinimized_impl() override
+	{
+		LOG("Window minimized");
+		return true;
+	}
+	bool onWindowMaximized_impl() override
+	{
+		LOG("Window maximized");
+		return true;
+	}
+	void onGainedMouseFocus_impl() override
+	{
+		LOG("Window gained mouse focus");
+	}
+	void onLostMouseFocus_impl() override
+	{
+		LOG("Window lost mouse focus");
+	}
+	void onGainedKeyboardFocus_impl() override
+	{
+		LOG("Window gained keyboard focus");
+	}
+	void onLostKeyboardFocus_impl() override
+	{
+		LOG("Window lost keyboard focus");
+		windowShouldClose_Global = true;
+	}
+
+	void onMouseConnected_impl(core::smart_refctd_ptr<nbl::ui::IMouseEventChannel>&& mch) override
+	{
+		LOG("A mouse has been connected");
+	}
+	void onMouseDisconnected_impl(nbl::ui::IMouseEventChannel* mch) override
+	{
+		LOG("A mouse has been disconnected");
+	}
+	void onKeyboardConnected_impl(core::smart_refctd_ptr<nbl::ui::IKeyboardEventChannel>&& kbch) override
+	{
+		LOG("A keyboard has been connected");
+	}
+	void onKeyboardDisconnected_impl(nbl::ui::IKeyboardEventChannel* mch) override
+	{
+		LOG("A keyboard has been disconnected");
+	}
+};
+
+struct alignas(256) UniformBufferObject
+{
+	float r, g, b, a;
+};
 
 int main()
 {
+	constexpr uint32_t WIN_W = 800u;
+	constexpr uint32_t WIN_H = 600u;
+	constexpr uint32_t MAX_SWAPCHAIN_IMAGE_COUNT = 16u;
+
+	auto system = CommonAPI::createSystem(); // Todo(achal): Need to get rid of this
+
+	auto winManager = core::make_smart_refctd_ptr<nbl::ui::CWindowManagerWin32>();
+
+	nbl::ui::IWindow::SCreationParams params;
+	params.callback = nullptr;
+	params.width = WIN_W;
+	params.height = WIN_H;
+	params.x = 0;
+	params.y = 0;
+	params.system = core::smart_refctd_ptr(system);
+	params.flags = nbl::ui::IWindow::ECF_NONE;
+	params.windowCaption = "02.ComputeShader";
+	params.callback = core::make_smart_refctd_ptr<DemoEventCallback>();
+	auto window = winManager->createWindow(std::move(params));
+
+	core::smart_refctd_ptr<video::CVulkanConnection> apiConnection =
+		video::CVulkanConnection::create(core::smart_refctd_ptr(system), 0, "02.ComputeShader", true);
+
+	core::smart_refctd_ptr<video::CSurfaceVulkanWin32> surface =
+		video::CSurfaceVulkanWin32::create(core::smart_refctd_ptr(apiConnection),
+			core::smart_refctd_ptr<ui::IWindowWin32>(static_cast<ui::IWindowWin32*>(window.get())));
```

Comment:
use and extend common API for (Example 1 already shows the user how to do this):
- system
- window
- api connection
- physical device retrieval
- logical device creation
- queue retrieval
- swapchain creationg and retrieval

Reply:
Resolved